File: app/src/main/java/org/stypox/dicio/skills/lyrics/LyricsInfo.java
package org.stypox.dicio.skills.lyrics;

import static org.stypox.dicio.Sections.getSection;
import static org.stypox.dicio.Sections.isSectionAvailable;
import static org.stypox.dicio.SectionsGenerated.lyrics;

import androidx.annotation.Nullable;
import androidx.preference.PreferenceFragmentCompat;

import org.stypox.dicio.R;
import org.dicio.skill.Skill;
import org.dicio.skill.SkillContext;
import org.dicio.skill.SkillInfo;
import org.dicio.skill.chain.ChainSkill;
import org.dicio.skill.standard.StandardRecognizer;

public class LyricsInfo extends SkillInfo {

    public LyricsInfo() {
        super("lyrics", R.string.skill_name_lyrics, R.string.skill_sentence_example_lyrics,
                R.drawable.ic_music_note_white, false);
    }

    @Override
    public boolean isAvailable(final SkillContext context) {
        return isSectionAvailable(lyrics);
    }

    @Override
    public Skill build(final SkillContext context) {
        return new ChainSkill.Builder()
                .recognize(new StandardRecognizer(getSection(lyrics)))
                .process(new GeniusProcessor())
                .output(new LyricsOutput());
    }

    @Nullable
    @Override
    public PreferenceFragmentCompat getPreferenceFragment() {
        return null;
    }
}

File: app/src/main/kotlin/org/stypox/dicio/skills/lyrics/LyricsInfo.kt
package org.stypox.dicio.skills.lyrics

import android.content.Context
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.MusicNote
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.vector.rememberVectorPainter
import org.dicio.skill.context.SkillContext
import org.dicio.skill.skill.Skill
import org.dicio.skill.skill.SkillInfo
import org.stypox.dicio.R
import org.stypox.dicio.Sections
import org.stypox.dicio.SectionsGenerated.lyrics
import org.stypox.dicio.sentences.Sentences

object LyricsInfo : SkillInfo("lyrics") {
    override fun name(context: Context) =
        context.getString(R.string.skill_name_lyrics)

    override fun sentenceExample(context: Context) =
        context.getString(R.string.skill_sentence_example_lyrics)

    @Composable
    override fun icon() =
        rememberVectorPainter(Icons.Default.MusicNote)

    override fun isAvailable(ctx: SkillContext): Boolean {
        return Sentences.Lyrics[ctx.locale.language] != null
    }

    override fun build(ctx: SkillContext): Skill<*> {
        return LyricsSkill(LyricsInfo, Sentences.Lyrics[ctx.locale.language]!!)
    }
}
